### PR TITLE
#288 Add Contributions Policy

### PR DESCRIPTION
**Pull request by timyhac**
* Contains a contributions Policy
* Includes an example of a PLC Fake, referred to in the debugging guidance.

### CODE REVIEW

File: src/Examples/CSharp DotNetCore/ExampleSimulator.cs [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+                };
+            }
+
+            public int Counter { get; internal set; }
+
+            public async Task ResetCounter()
+            {
+                // The PLC is expected to have some logic that sets the ResetFlag from true back to false after resetting the counter
+                await ResetTag.WriteAsync(true);
+                await ReceiveData();
```
Comment (jkoplo) [inline src/Examples/CSharp DotNetCore/ExampleSimulator.cs:128]: I know this is just a throw away example, but it bothers me that this is a race condition...

**Comment by timyhac** [comment]
@jkoplo - any comments around the Contributions Policy?

File: src/Examples/CSharp DotNetCore/ExampleSimulator.cs (hunk 1 above)
Comment (jkoplo) [reply src/Examples/CSharp DotNetCore/ExampleSimulator.cs:128]: Well, in this case you're writing `true` to a PLC tag to presumably signal the PLC to reset a counter. And then you're reading the value of that counter with `ReceiveData`.  Depending how fast the PLC is scanning and how much network delay, etc you will either read back the value of the counter before the PLC processes the `ResetTag` trigger or you'll read it back after it processes `ResetTag`...

Like I said for the purpose of the example it's probably fine. In real life I'd probably set `ResetTag` to `true`, poll the tag until the PLC processes and sets it back to false (assuming PLC logic and order of operations), and then read the value of `CounterTag`.

**Comment by jkoplo** [comment]
And I never got back to you on the contribution policy. 👍

Looks good to me.

File: src/Examples/CSharp DotNetCore/ExampleSimulator.cs (hunk 1 above)
Comment (timyhac) [reply src/Examples/CSharp DotNetCore/ExampleSimulator.cs:128]: I see! Good points

**Pull request merged**